### PR TITLE
eslint fixes and eslint config fixes

### PR DESCRIPTION
**Pull request by rkofman**

### CODE REVIEW

File: .eslintignore
Comment (rkofman) [inline .eslintignore:1]: This file is deprecated in favor of an ignores property in eslint config.

File: .eslintrc.cjs
Comment (rkofman) [inline .eslintrc.cjs:1]: No idea what this file was doing, tbh.

File: e2e/test.e2e.ts
Comment (rkofman) [inline e2e/test.e2e.ts:1]: lint errors related to await issues. Only promises need awaiting, and async promises _must_ be awaited.

File: eslint.config.js [hunk 1]
```diff
@@ -1,22 +1,32 @@
...
+// @ts-expect-error "no types for eslint-plugin-storybook"
+import storybook from 'eslint-plugin-storybook'
 import globals from 'globals'
 
-/** @type {import('eslint').Linter.Config[]} */
-export default [
+export default tseslint.config(
   js.configs.recommended,
-  ...ts.configs.recommended,
+  ...ts.configs.recommendedTypeChecked,
```
Comment (rkofman) [inline eslint.config.js]: linting for type issues as well as regular issues.

File: eslint.config.js [hunk 2]
```diff
@@ -1,22 +1,32 @@
+// @ts-check
 import js from '@eslint/js'
 import ts from 'typescript-eslint'
+import tseslint from 'typescript-eslint'
```
Comment (rkofman) [inline eslint.config.js:3]: uhh; I should probably combine these two imports, this is silly.
- [x] TODO: fix this.

File: eslint.config.js [hunk 3]
```diff
@@ -1,22 +1,32 @@
...
-export default [
+export default tseslint.config(
   js.configs.recommended,
-  ...ts.configs.recommended,
+  ...ts.configs.recommendedTypeChecked,
   ...svelte.configs['flat/recommended'],
   prettier,
   ...svelte.configs['flat/prettier'],
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-argument, @typescript-eslint/no-unsafe-member-access
+  ...storybook.configs['flat/recommended'],
```
Comment (rkofman) [inline eslint.config.js:17]: not that we're using storybook yet, but let's include linting for it.

File: eslint.config.js [hunk 4]
```diff
@@ -1,22 +1,32 @@
...
   {
     languageOptions: {
       globals: {
         ...globals.browser,
         ...globals.node,
       },
+      parserOptions: {
+        projectService: true,
+        tsconfigRootDir: import.meta.dirname,
+        extraFileExtensions: ['svelte'],
```
Comment (rkofman) [inline eslint.config.js:27]: required for type-aware linting.

File: eslint.config.js [hunk 5]
```diff
@@ -28,6 +38,22 @@ export default [
...
+      'src-ui/stories/',
+      'src-ui/lib/generated/',
+      '!.storybook',
+    ],
   },
-]
+  {
+    rules: {
+      'require-await': 'off',
+      '@typescript-eslint/require-await': 'error',
```
Comment (rkofman) [inline eslint.config.js:56]: weird way to say we want type-aware lints for await.

File: src-ui/lib/components/Versions.svelte
Comment (rkofman) [inline src-ui/lib/components/Versions.svelte:1]: Just leftover crud from electron days.

File: tsconfig.json [hunk 6]
```diff
@@ -9,9 +9,21 @@
...
+    "./src-ui/**/*.js",
+    "./src-ui/**/*.ts",
+    "./src-ui/**/*.svelte",
+    "./.storybook/**/*.js",
+    "./.storybook/**/*.ts",
+    "./.storybook/**/*.svelte",
+    "./*.config.js",
+    "./*.config.ts",
+    "./vitest.diff.ts"
+  ]
```
Comment (rkofman) [inline tsconfig.json:26]: needed to extend the `include` list from .svelte-kit/tsconfig.json; and there is no way to do so without overriding the whole thing.

File: tsconfig.node.json
Comment (rkofman) [inline tsconfig.node.json:1]: yeah, I don't know how half these useless competing generated configs made it in. My bad.

File: eslint.config.js (hunk 2 above)
Comment (SilverMaiden) [reply eslint.config.js:3]: 😂 <3

File: eslint.config.js (hunk 3 above)
Comment (SilverMaiden) [reply eslint.config.js:17]: We should be getting to that pretty soon

**Review by SilverMaiden — APPROVED** [review]
🎉

**Pull request merged**